aws/aws-parallelcluster
# [PCAPI] Add resources for local testing of PCAPI with Docker

### Description of changes
Add resources for local testing of PCAPI with Docker. Those utilities were removed in https://github.com/aws/aws-parallelcluster/pull/6351 because, since PC 3.6.0 we moved the PCAPI lambda from Docker to Lambda Layer.

However, those resources are useful to test PCAPI locally with Docker and it's worth it to have them back.

Other minor changes:
1. removed deprecated instructions about publishing the Docker image to ECr Public, as we do not publish it anymore.
2. extended the instructions to run the container so that AWS credentials can be injected in the container.

### Tests
1. Built the PCAPI locally: `bash docker-build.sh`
2. Run the PCAPI container locally as a Flask application: `docker run -p 8080:8080 -v ~/.aws:/root/.aws:ro --entrypoint python pcluster-lambda -m pcluster.api.flask_app`
3. Execute a ListClusters call with PostMan, returning the expected value 

### References
* Link to impacted open issues.
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
